OnlineStore/Controllers/AuthController.cs

```diff
@@ -0,0 +1,55 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using OnlineStore.BLL.DTOs.Register;
+using OnlineStore.BLL.Services.Interfaces;
+
+namespace OnlineStore.WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly IAuthService _authService;
+
+        public AuthController(IAuthService authService)
+        {
+            _authService = authService;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(UserRegisterDto userRegisterDto)
+        {
+            try
+            {
+                var result = await _authService.RegisterAsync(userRegisterDto);
+                return Ok(result);
+            }
+            catch (ValidationException ex)
+            {
+                return BadRequest(new { Errors = ex.Errors });
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(new { Error = ex.Message });
+            }
```

Comment:
no need to write a try catch, for this you have written a global exception handler